web/src/components/UserInformationForm.tsx

```diff
@@ -1,82 +1,219 @@
-import FormInput from "@components/FormInput.tsx";
+import FormInput from "../components/FormInput";
 import { useState } from "react";
+import axios from "axios";
 
 function UserInformationForm() {
   const [values, setValues] = useState({
     firstName: "",
     lastName: "",
     university: "",
-    graudationYear: "",
+    graduationYear: "",
     UPI: "",
     studentID: "",
     studyOption: "",
+    residency: "",
+    duration: "",
+  });
+
+  const [errors, setErrors] = useState({
+    firstName: "",
+    lastName: "",
+    university: "",
+    graduationYear: "",
+    duration: "",
   });
 
   const onChange = (e: React.ChangeEvent<HTMLInputElement>) => {
-    // Need to do validation here
     setValues({ ...values, [e.target.name]: e.target.value });
-    console.log(values);
+    setErrors({ ...errors, [e.target.name]: "" });
   };
 
   const residencyOptions = ["International", "Domestic"];
   const paymentOptions = ["One Semester ($8)", "Two Semesters ($15)"];
 
-  function handleSubmit(e: React.FormEvent<HTMLFormElement>) {
+  const validate = () => {
```

Comment:
I'll have to check with Naren but I think there is a better way to handle validation. Not to only make sure that it is entered. but also to make sure we only accept characters that are allowed